lib/QoreNamespace.cpp

```diff
@@ -2853,57 +3022,279 @@ QoreNamespace* qore_ns_private::parseFindLocalNamespace(const char* nname) {
 }
 
 StaticSystemNamespace::~StaticSystemNamespace() {
-   ExceptionSink xsink;
-   priv->deleteData(true, &xsink);
-   priv->purge();
+    ExceptionSink xsink;
+    priv->deleteData(true, &xsink);
+    priv->purge();
 }
 
 QoreNamespaceIterator::QoreNamespaceIterator(QoreNamespace* ns) : priv(new QorePrivateNamespaceIterator(qore_ns_private::get(*ns))) {
 }
 
 bool QoreNamespaceIterator::next() {
-   return priv->next();
+    return priv->next();
 }
 
 QoreNamespace* QoreNamespaceIterator::operator->() {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 QoreNamespace* QoreNamespaceIterator::operator*() {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 const QoreNamespace* QoreNamespaceIterator::operator->() const {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 const QoreNamespace* QoreNamespaceIterator::operator*() const {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 QoreNamespace* QoreNamespaceIterator::get() {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 const QoreNamespace* QoreNamespaceIterator::get() const {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 QoreNamespaceConstIterator::QoreNamespaceConstIterator(const QoreNamespace* ns) : priv(new QorePrivateNamespaceIterator(const_cast<qore_ns_private*>(qore_ns_private::get(*ns)))) {
 }
 
 bool QoreNamespaceConstIterator::next() {
-   return priv->next();
+    return priv->next();
 }
 
 const QoreNamespace* QoreNamespaceConstIterator::operator->() const {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 const QoreNamespace* QoreNamespaceConstIterator::operator*() const {
-   return priv->get()->ns;
+    return priv->get()->ns;
 }
 
 const QoreNamespace* QoreNamespaceConstIterator::get() const {
-   return priv->get()->ns;
+    return priv->get()->ns;
+}
+
+class qore_namespace_namespace_iterator {
+public:
+    DLLLOCAL qore_namespace_namespace_iterator(const qore_ns_private* ns) :
+        ns(ns), i(ns->nsl.nsmap.end()) {
+    }
+
+    DLLLOCAL bool next() {
+        if (i == ns->nsl.nsmap.end()) {
+            i = ns->nsl.nsmap.begin();
+        }
+        else {
+            ++i;
+        }
+        return (i != ns->nsl.nsmap.end());
```

Comment:
this will (almost) always return true, right? because when it reaches end, it will go over again except for the case when begin==end. It's not documented that the iterator is cyclic, why do we need it?